Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -857,8 +247,14 @@ class SettingsTableViewController: UITableViewController {
 
     override func viewWillAppear(animated: Bool) {
         super.viewWillAppear(animated)
+
+        settings = generateSettings()
+
         NSNotificationCenter.defaultCenter().addObserver(self, selector: "SELsyncDidChangeState", name: NotificationProfileDidStartSyncing, object: nil)
         NSNotificationCenter.defaultCenter().addObserver(self, selector: "SELsyncDidChangeState", name: NotificationProfileDidFinishSyncing, object: nil)
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: "SELfirefoxAccountDidChange", name: NotificationFirefoxAccountChanged, object: nil)
```

Comment:
In this case it seems like its only specific to the app settings but I feel like this has the potential to be used in many settings screens. Generally settings are tied to users profiles so I can see this being something we want. The callbacks right now only reload the table views which I think is OK but if we get more complicated logic for these callbacks down the road I'd say we break this out into an Account-based settings base class.